plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/TrinoGlueCatalog.java

```diff
@@ -1108,6 +1108,8 @@ private Optional<ConnectorMaterializedViewDefinition> createMaterializedViewDefi
             // - org.apache.iceberg.exceptions.NotFoundException when accessing manifest file
             // - other failures when reading storage table's metadata files
             // Retry, as we're catching broadly.
+            deleteTable(viewName.getSchemaName(), viewName.getTableName());
+            deleteTable(storageSchema, storageTable);
```

Comment:
Add a test for this situation to verify whether the view and the storage table don't exist anymore after the query in case of dealing with such an exception